test/tests/mfem/submeshes/domain_submesh_transfer.i

```diff
@@ -0,0 +1,108 @@
+[Mesh]
+  type = MFEMMesh
+  file = ../mesh/cylinder-hex-q2.gen
+[]
+
+[Problem]
+  type = MFEMProblem
+[]
+
+[SubMeshes]
+  [wire]
+    type = MFEMDomainSubMesh
+    block = 1
+  []
+[]
+
+[FESpaces]
+  [SubMeshH1FESpace]
+    type = MFEMScalarFESpace
+    fec_type = H1
+    fec_order = FIRST
+    submesh = wire
+  []
+  [H1FESpace]
+    type = MFEMScalarFESpace
+    fec_type = H1
+    fec_order = FIRST
+  []
+[]
+
+[Variables]
+  [submesh_potential]
+    type = MFEMVariable
+    fespace = SubMeshH1FESpace
+  []
+[]
+
+[AuxVariables]
+  [potential]
+    type = MFEMVariable
+    fespace = H1FESpace
+  []
+[]
+
+[BCs]
+  [bottom]
+    type = MFEMScalarDirichletBC
+    variable = submesh_potential
+    boundary = '1'
+    value = 1.0
+  []
+  [low_terminal]
+    type = MFEMScalarDirichletBC
+    variable = submesh_potential
+    boundary = '2'
+    value = 0.0
+  []
+[]
+
+[FunctorMaterials]
+  [Substance]
+    type = MFEMGenericConstantFunctorMaterial
+    prop_names = diffusivity
+    prop_values = 1.0
+  []
+[]
+
+[Kernels]
+  [diff]
+    type = MFEMDiffusionKernel
+    variable = submesh_potential
+    coefficient = diffusivity
+  []
+[]
+
+[Preconditioner]
+  [boomeramg]
+    type = MFEMHypreBoomerAMG
+  []
+[]
+
+[Solver]
+  type = MFEMHypreGMRES
+  preconditioner = boomeramg
+  l_tol = 1e-16
+  l_max_its = 1000
+[]
+
+[Executioner]
+  type = MFEMSteady
+  device = cpu
+[]
+
+[Transfers]
+  [submesh_transfer]
+    type = MFEMSubMeshTransfer
+    from_variable = submesh_potential
+    to_variable = potential
```

Comment:
so how does this transfer work? How do we fill the parent domain from just a part of it? What algorithm? E.g. thinking in other transfer terms like evaluating the source solution (can't be that because presumably the solution is a restriction of the parent domain), nearest node (doubt MFEM would do something so primeval :smile:), etc.